dll/win32/shell32/CDefaultContextMenu.cpp

```diff
@@ -282,26 +287,29 @@ void CDefaultContextMenu::AddStaticEntry(const HKEY hkeyClass, const WCHAR *szVe
         pEntry->hkClass = hkeyClass;
     }
 
-    if (!wcsicmp(szVerb, L"open"))
+    lres = RegQueryValueEx(hShellKey, NULL, 0, &pdwType, (LPBYTE)wszDefault, &dwSize);
+	
+    if((!wcsicmp(szVerb, wszDefault) && lres == ERROR_SUCCESS) || (!wcsicmp(szVerb, L"open") && lres != ERROR_SUCCESS))
     {
-        /* open verb is always inserted in front */
+        /* open verb is always inserted in front, unless the default value of shell entry is set*/
         pEntry->pNext = m_pStaticEntries;
         m_pStaticEntries = pEntry;
     }
     else if (pLastEntry)
         pLastEntry->pNext = pEntry;
     else
         m_pStaticEntries = pEntry;
+	
+    RegCloseKey(hShellKey);
 }
 
 void CDefaultContextMenu::AddStaticEntriesForKey(HKEY hKey)
 {
     WCHAR wszName[40];
     DWORD cchName, dwIndex = 0;
     HKEY hShellKey;
-
-    LRESULT lres = RegOpenKeyExW(hKey, L"shell", 0, KEY_READ, &hShellKey);
-    if (lres != STATUS_SUCCESS)
+	
+    if (RegOpenKeyExW(hKey, L"shell", 0, KEY_READ, &hShellKey) != STATUS_SUCCESS)
```

Comment:
This seems less readable to me. Please leave it as it was.
Although changing STATUS_SUCCESS to ERROR_SUCCESS certainly would be an improvement :p